### PR TITLE
SPKAC: Allow the message digest to be specified explicitly

### PR DESCRIPTION
**Pull request by minfrin**
Right now, the spkac command hardcodes the message digest to MD5.

This patch adds the -md option following the pattern of other commands where a message digest is specified.

This brings openssl's spkac support in compliant with https://tools.ietf.org/html/rfc6151.

- [X] documentation is added or updated

### CODE REVIEW

File: doc/man1/openssl-spkac.pod.in [hunk 1]
```diff
@@ -93,6 +94,11 @@ SPKAC is being created).
 Output the public key of an SPKAC (not used if an SPKAC is
 being created).
 
+=item B<-md>
+
+The message digest used for signing. Historically the default value has
+been MD5, but SHA256 is recommended.
+
```
Comment (beldmit) [inline doc/man1/openssl-spkac.pod.in]: Shouldn't it be specified by default in code too?

Comment (minfrin) [reply doc/man1/openssl-spkac.pod.in]: Happy to specify sha256 by default, openssl happy with the change in default behaviour?

**Comment by richsalz** [comment]
I am curious how you are using SPKAC's?

**Comment by minfrin** [comment]
> I am curious how you are using SPKAC's?

We (https://redwax.eu) are putting together a draft RFC to standardise SPKAC (and other historic) protocols so there is a formal record of their use, as well as a formal record of their shortcomings.

One thing we want to do is to fix shortcomings in their implementations, the use of md5 in this case is one of them.

**Comment by richsalz** [comment]
Are you using SPKAC or is this just a documentation effort?  See #1177 for some discussion.

**Comment by mattcaswell** [comment]
Close/reopen to kick CLA bot

**Comment by mattcaswell** [comment]
Try again...

**Comment by t8m** [comment]
This was already added by another PR. Closing.

**Comment by minfrin** [comment]
For the record, this PR was committed here to v3.0.0:

https://github.com/openssl/openssl/commit/e1a77f9cffbd7f8642ff900a3e5b7c81e8c26fb7
https://github.com/openssl/openssl/commit/81743ed9d737d415a43aaf0259616dd007a9e3a4

**Pull request closed without merging**